controlplane/kubeadm/internal/controllers/status.go

```diff
@@ -423,6 +424,139 @@ func setRemediatingCondition(ctx context.Context, kcp *controlplanev1.KubeadmCon
+				v1beta2conditions.IsTrue(machine, controlplanev1.KubeadmControlPlaneMachineSchedulerPodHealthyV1Beta2Condition) {
+				k8sControlPlaneHealthy++
+			}
+			continue
+		}
+
+		// Otherwise, etcd is managed.
+		// In this case, when looking at the k8s control plane we should consider how kubeadm layouts control plane components,
+		// and more specifically:
+		// - API server on one machine only connect to the local etcd member
+		// - ControllerManager and scheduler on a machine connect to the local API server (not to the control plane endpoint)
+		// As a consequence, we consider the K8s control plane on this machine healthy only if everything is healthy.
+
+		if v1beta2conditions.IsTrue(machine, controlplanev1.KubeadmControlPlaneMachineEtcdMemberHealthyV1Beta2Condition) {
+			etcdMembersHealthy++
+		}
+
+		if v1beta2conditions.IsTrue(machine, controlplanev1.KubeadmControlPlaneMachineAPIServerPodHealthyV1Beta2Condition) &&
+			v1beta2conditions.IsTrue(machine, controlplanev1.KubeadmControlPlaneMachineControllerManagerPodHealthyV1Beta2Condition) &&
+			v1beta2conditions.IsTrue(machine, controlplanev1.KubeadmControlPlaneMachineSchedulerPodHealthyV1Beta2Condition) &&
+			v1beta2conditions.IsTrue(machine, controlplanev1.KubeadmControlPlaneMachineEtcdMemberHealthyV1Beta2Condition) &&
+			v1beta2conditions.IsTrue(machine, controlplanev1.KubeadmControlPlaneMachineEtcdPodHealthyV1Beta2Condition) {
+			k8sControlPlaneHealthy++
+		}
+	}
+
+	if kcp.DeletionTimestamp.IsZero() &&
+		(!etcdIsManaged || etcdMembersHealthy >= etcdQuorum) &&
+		k8sControlPlaneHealthy >= 1 &&
+		v1beta2conditions.IsTrue(kcp, controlplanev1.KubeadmControlPlaneCertificatesAvailableV1Beta2Condition) {
+		v1beta2conditions.Set(kcp, metav1.Condition{
+			Type:   controlplanev1.KubeadmControlPlaneAvailableV1Beta2Condition,
+			Status: metav1.ConditionTrue,
+			Reason: controlplanev1.KubeadmControlPlaneAvailableV1Beta2Reason,
+		})
+		return
+	}
+
+	messages := []string{}
+	if !kcp.DeletionTimestamp.IsZero() {
+		messages = append(messages, "Control plane metadata.deletionTimestamp is set")
+	}
+
+	if !v1beta2conditions.IsTrue(kcp, controlplanev1.KubeadmControlPlaneCertificatesAvailableV1Beta2Condition) {
+		messages = append(messages, "Control plane certificates are not available")
+	}
+
+	if etcdIsManaged && etcdMembersHealthy < etcdQuorum {
+		switch etcdMembersHealthy {
+		case 0:
+			messages = append(messages, fmt.Sprintf("There are no healthy etcd member, at least %d required", etcdQuorum))
+		case 1:
+			messages = append(messages, fmt.Sprintf("There is 1 healthy etcd member, at least %d required", etcdQuorum))
+		default:
+			messages = append(messages, fmt.Sprintf("There are %d healthy etcd members, at least %d required", etcdMembersHealthy, etcdQuorum))
+		}
+	}
+
+	if k8sControlPlaneHealthy < 1 {
+		messages = append(messages, "There are no healthy control plane instances, at least 1 required")
```

Comment:
```suggestion
		messages = append(messages, "There are no healthy control plane Machines, at least 1 required")
```

An alternative
"There are no Machines with healthy control plane components, at least 1 required"

maybe, not sure. Not sure if we should introduce new wording with "instance"